common/components/ConfirmationModalTemplate/components/Body/Body.tsx

```diff
@@ -0,0 +1,119 @@
+import { Addresses } from './components/Addresses';
+import { Amounts } from './components/Amounts';
+import { Details } from './components/Details';
+import React from 'react';
+import { SerializedTransaction } from 'components/renderCbs';
+import { makeTransaction, getTransactionFields } from 'libs/transaction';
+import { connect } from 'react-redux';
+import { AppState } from 'reducers';
+import { getFrom, getUnit, getDecimal, getTo } from 'selectors/transaction';
+import { getNetworkConfig, getNodeConfig } from 'selectors/config';
+import { getTransactionFee } from 'libs/transaction/utils/ether';
+import { Wei, TokenValue } from 'libs/units';
+import ERC20 from 'libs/erc20';
+import './Body.scss';
+
+interface State {
+  showDetails: boolean;
+}
+
+interface Props {
+  rates: AppState['rates']['rates'];
+  from: string;
+  to: string;
+  unit: string;
+  node: AppState['config']['node'];
+  network: AppState['config']['network'];
+  decimal: number;
+}
+
+class BodyClass extends React.Component<Props, State> {
+  public state: State = {
+    showDetails: false
+  };
+
+  public toggleDetails = () => {
+    this.setState({
+      showDetails: !this.state.showDetails
+    });
+  };
+
+  public render() {
+    const { rates, from, to, unit, node, network, decimal } = this.props;
+    const { showDetails } = this.state;
+    const networkUnit = network.unit;
+    return (
+      <SerializedTransaction
+        withSerializedTransaction={serializedTransaction => {
+          const transactionInstance = makeTransaction(serializedTransaction);
+          const rawTx = getTransactionFields(transactionInstance);
+          const { value, gasPrice, gasLimit, data, nonce, chainId } = rawTx;
+          const isToken = !(unit === 'ether' || unit === 'ETH');
```

Comment:
This will break on non-ether unit networks. Instead it should be checking if the supplied unit is a non-network unit `!network.unit`.  This is also more suited to be a selector rather than in-component